src/IW/Sync/Update.hs

```diff
@@ -4,23 +4,38 @@ data and insert it into the database.
 -}
 
 module IW.Sync.Update
-       ( syncRepos
+       ( syncCache
        ) where
 
 import Control.Monad.IO.Unlift (MonadUnliftIO)
-import UnliftIO.Async (mapConcurrently_)
+import UnliftIO.Async (async, mapConcurrently_, wait)
 
-import IW.App (WithError)
+import IW.App (AppErrorType, WithError, catchError)
 import IW.Core.Repo (Repo (..))
-import IW.Db (WithDb, upsertRepos, updateRepoCategories)
+import IW.Db (WithDb, upsertRepos, updateRepoCategories, upsertIssues)
 import IW.Effects.Cabal (MonadCabal (..), getCabalCategories)
-import IW.Sync.Search (searchAllHaskellRepos)
+import IW.Sync.Search (searchAllHaskellRepos, searchAllHaskellIssues)
 import IW.Time (getToday)
 
 
--- | This function fetches all repos from the GitHub API, downloads their @.cabal@ files,
--- and upserts them into the database.
-syncRepos
+-- | Loop for keeping the database up to date with the latest GitHub data.
+syncCache
+    :: forall env m.
+       ( MonadCabal m
+       , MonadUnliftIO m
+       , WithDb env m
+       , WithLog env m
+       , WithError m
+       )
+    => Integer
+    -> m ()
+syncCache interval = forever $ syncWithGithub interval `catchError` syncErrHandler
+  where
+    syncErrHandler :: AppErrorType -> m a
+    syncErrHandler = undefined
```

Comment:
@chshersh @vrom911 Okay so once I implement this handler I think this PR is done. How should we handle the errors? We don't want it to stop syncing if an error is encountered, so do I have to think of some way to resume the sync process from where the error was thrown? Or is it simpler than that?

Reply:
I will think about the problem :thinking: